File: ANewFolder/MyCards-master(1)/MyCards-master/app/src/main/java/edu/univdhaka/cse/cse2216/mycards/java/repository/DatabaseCardRepository.java
package edu.univdhaka.cse.cse2216.mycards.java.repository;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.support.annotation.NonNull;

import java.util.ArrayList;
import java.util.List;

import edu.univdhaka.cse.cse2216.mycards.java.domain.Card;
import edu.univdhaka.cse.cse2216.mycards.java.utils.DbHelper;

public class DatabaseCardRepository implements CardRepository {

    private Context context;

    public DatabaseCardRepository(Context context) {
        this.context = context;
    }

    @Override
    public List<Card> listCards() {
        List<Card> cards = new ArrayList<>();

        try (SQLiteDatabase db = new DbHelper(context).getReadableDatabase();
             Cursor cursor = db.query("cards", null, null, null, null, null, null)) {

            while (cursor.moveToNext()) {
                cards.add(
                        new Card(
                                cursor.getString(cursor.getColumnIndexOrThrow("type")),
                                cursor.getString(cursor.getColumnIndexOrThrow("bank_name")),
                                cursor.getString(cursor.getColumnIndexOrThrow("number")),
                                cursor.getString(cursor.getColumnIndexOrThrow("cardholder_name")),
                                cursor.getString(cursor.getColumnIndexOrThrow("expiry_date")),
                                cursor.getInt(cursor.getColumnIndexOrThrow("cvv")),
                                cursor.getInt(cursor.getColumnIndexOrThrow("_id"))
                        )
                );
            }
        }

        return cards;
    }

    @Override
    public void addCard(Card card) {
        ContentValues values = getContentValues(card);

        try (SQLiteDatabase db = new DbHelper(context).getWritableDatabase()) {
            db.insertOrThrow("cards", null, values);
        }
    }

    @Override
    public void updateCard(Card card) {
        ContentValues values = getContentValues(card);

        try (SQLiteDatabase db = new DbHelper(context).getWritableDatabase()) {
            db.update("cards", values, "_id = ?", new String[]{Integer.toString(card.getId())});
        }
    }

    @Override
    public void deleteCard(Card card) {
        try (SQLiteDatabase db = new DbHelper(context).getWritableDatabase()) {
            db.delete("cards", "_id = ?", new String[]{Integer.toString(card.getId())});
        }
    }

    @NonNull
    private ContentValues getContentValues(Card card) {
        ContentValues values = new ContentValues();
        values.put("number", card.getNumber());
        values.put("type", card.getType());
        values.put("bank_name", card.getBankName());
        values.put("cardholder_name", card.getCardholderName());
        values.put("expiry_date", card.getExpiryDate());
        values.put("cvv", card.getCvv());

        return values;
    }
}

File: ANewFolder/MyCards-master (2)/app/src/main/java/edu/univdhaka/cse/cse2216/mycards/kotlin/repository/DatabaseCardRepository.kt
package edu.univdhaka.cse.cse2216.mycards.kotlin.repository

import android.content.ContentValues
import android.content.Context
import edu.univdhaka.cse.cse2216.mycards.kotlin.domain.Card
import edu.univdhaka.cse.cse2216.mycards.kotlin.utils.DbHelper

class DatabaseCardRepository(private val context: Context) : CardRepository {

    override fun listCards(): List<Card> {
        val cards = ArrayList<Card>()

        DbHelper(context).readableDatabase.use { db ->
            db.query("cards", null, null, null, null, null, null).use { cursor ->

                while (cursor.moveToNext()) {
                    cards.add(
                            Card(
                                    cursor.getString(cursor.getColumnIndexOrThrow("type")),
                                    cursor.getString(cursor.getColumnIndexOrThrow("bank_name")),
                                    cursor.getString(cursor.getColumnIndexOrThrow("number")),
                                    cursor.getString(cursor.getColumnIndexOrThrow("cardholder_name")),
                                    cursor.getString(cursor.getColumnIndexOrThrow("expiry_date")),
                                    cursor.getInt(cursor.getColumnIndexOrThrow("cvv")),
                                    cursor.getInt(cursor.getColumnIndexOrThrow("_id"))
                            )
                    )
                }
            }
        }

        return cards
    }

    override fun listCardsAsync(resultListener: (List<Card>) -> Unit) {
        resultListener(listCards())
    }

    override fun addCard(card: Card) {
        val values = getContentValues(card)

        DbHelper(context).writableDatabase.use { db -> db.insertOrThrow("cards", null, values) }
    }

    override fun updateCard(card: Card) {
        val values = getContentValues(card)

        DbHelper(context).writableDatabase.use { db -> db.update("cards", values, "_id = ?", arrayOf(Integer.toString(card.id))) }
    }

    override fun deleteCard(card: Card) {
        DbHelper(context).writableDatabase.use { db -> db.delete("cards", "_id = ?", arrayOf(Integer.toString(card.id))) }
    }

    private fun getContentValues(card: Card): ContentValues {
        val values = ContentValues()
        values.put("number", card.number)
        values.put("type", card.type)
        values.put("bank_name", card.bankName)
        values.put("cardholder_name", card.cardholderName)
        values.put("expiry_date", card.expiryDate)
        values.put("cvv", card.cvv)

        return values
    }
}
